### PR TITLE
EVG-7437: add distro and host buttons to restart Jasper

### PR DESCRIPTION
**Pull request by Kimchelly**
JIRA: https://jira.mongodb.org/browse/EVG-7437

This is a "push button, get new Jasper on host" UI feature.

* Jasper can be restarted for non-legacy hosts/distros from the UI (distro, host, and all hosts pages).
* Download latest version of Jasper binary in Jasper restart job. Gives us an easy way to ensure we have a way to easily update Jasper especially on static hosts.
* Fix some user data file permissions to allow overwriting credentials files.
* Increase frequency with which Jasper restart jobs get populated so we don't wait an extremely long time reprovisioning Jasper.
* Trivial fix to stop logging events when we clear nonexistent running tasks in the agent revision is old in next task endpoint.

### CODE REVIEW

File: model/event/host_event.go [hunk 1]
```diff
@@ -132,6 +133,10 @@ func LogHostAgentMonitorDeployFailed(hostId string, err error) {
 	LogHostEvent(hostId, EventHostAgentMonitorDeployFailed, HostEventData{Logs: err.Error()})
 }
 
+func LogHostJasperRestarting(hostId string, user string) {
```
Comment (bsamek) [inline model/event/host_event.go]: go style nit: `hostId, user string`

File: model/host/db.go [hunk 2]
```diff
@@ -460,6 +460,26 @@ func FindByShouldConvertProvisioning() ([]Host, error) {
...
+// restart their Jasper service.
+func FindByNeedsJasperRestart() ([]Host, error) {
+	return Find(db.Query(bson.M{
+		StatusKey:           bson.M{"$in": []string{evergreen.HostProvisioning, evergreen.HostRunning}},
+		RunningTaskKey:      bson.M{"$exists": false},
+		HasContainersKey:    bson.M{"$ne": true},
+		ParentIDKey:         bson.M{"$exists": false},
+		NeedsReprovisionKey: ReprovisionJasperRestart,
+		"$or": []bson.M{
+			{StartedByKey: bson.M{"$ne": evergreen.User}},
```
Comment (bsamek) [inline model/host/db.go]: This means that we plan to restart jasper on spawn hosts. Is that the intent?

File: model/host/db.go [hunk 3]
```diff
@@ -469,7 +489,6 @@ func NeedsReprovisioningLocked(currentTime time.Time) bson.M {
 	cutoffTime := currentTime.Add(-MaxLCTInterval)
 	return bson.M{
 		StatusKey:               evergreen.HostProvisioning,
-		StartedByKey:            evergreen.User,
```
Comment (bsamek) [inline model/host/db.go]: Same question about why we are including spawn hosts.

File: model/event/host_event.go (hunk 1 above)
Comment (Kimchelly) [reply model/event/host_event.go]: Done.

File: model/host/db.go (hunk 2 above)
Comment (Kimchelly) [reply model/host/db.go]: Yes, Jasper is designed to be restarted on all hosts. This only matters if the spawn host user happens to be running some script through Jasper and Jasper gets restarted at that exact second (which is something Jonathan wrote I think), but I honestly don't think it's worth handling.

File: model/host/db.go (hunk 3 above)
Comment (Kimchelly) [reply model/host/db.go]: Same response. Spawn hosts aren't immune to this requirement.

**Pull request merged**